beacon-chain/db/db.go

```diff
@@ -23,6 +23,7 @@ type BeaconDB struct {
 	DatabasePath string
 
 	// Beacon chain deposits in memory.
+	pendingDeposits     []*depositContainer
 	deposits     []*depositContainer
```

Comment:
We cache all historical deposits as well as pending deposits. Historical deposits will never be cleared out, but pending deposits will be cleared from the cache after insertion into the state.